### PR TITLE
introduce list of choices

### PR DESCRIPTION
**Pull request by mixilchenko**
- allow `is_choice` to work with `is_list`
- fix choice `default` default value from `None` to `MISSING`
  `None` default breaks `default_factory` handling
- remove redundant `test/test_choice.py::test_choice_with_dict_weird`
- add tests with lists of choices

### CODE REVIEW

File: test/test_choice.py
Comment (mixilchenko) [inline test/test_choice.py:65]: @lebrice I think this test was incorrect.
Type of `option` must be `Union[List[Base], float, Base]` because we specify here type of final attribute.

I've merged this test with `test_choice_with_dict` above.
If you disagree, I can fix it as you say

File: simple_parsing/wrappers/field_wrapper.py
Comment (lebrice) [inline simple_parsing/wrappers/field_wrapper.py:692]: Interesting, I didn't think of doing it that way.
I think it would ideally be better not to do it through changing `self._type`, because this indirectly depends on the other part of the code that constructs the `arg_options` dict that gets passed to `add_argument`.

For now you can leave this as-is, and I'll think about it some more.

File: test/test_choice.py
Comment (lebrice) [reply test/test_choice.py:65]: Sure, this is fine. Thanks!

**Review by lebrice — APPROVED** [review]
Hey @mixilchenko , thanks a lot for making a contribution! :smile: 

This looks good! Feel free to merge this, and I'll push a new tag soon.

Thanks again!

**Comment by mixilchenko** [comment]
@lebrice I don't have write access to this repo so I can't merge this request.
For me it's ok for owner to merge

**Pull request merged**
